BigQuery/src/Connection/Rest.php

```diff
@@ -419,4 +422,26 @@ public function testTableIamPermissions(array $args = [])
     {
         return $this->send('tables', 'testIamPermissions', $args);
     }
+
+    /**
+     * Creates a request wrapper and sets the HTTP Handler logger to the configured one.
+     *
+     * @param array $config
+     * @return RequestWrapper
+     */
+    private function getRequestWrapper(array $config): RequestWrapper
+    {
+        // Because we are setting a logger, we build a handler here instead of using the default
+        $config['httpHandler'] = $config['httpHandler'] ?? HttpHandlerFactory::build(logger: $config['logger']);
+        $config['restRetryListener'] = $this->getRetryListener();
+        return new RequestWrapper($config);
+    }
+
+    private function getRetryListener(): callable
```

Comment:
Can we add a php doc explaining what this function does? e.g. "The retry listener will get called in {@see ClassName} on retry. We need to supply a custom listener so that the "retryAttempt" is (whatever happens to it) by (whatever does that thing)"

This would help me as well because I am confused by the `retryAttempt` option - it doesn't appear to be a header, and it doesn't seem to show up anywhere else. Is it just a "call option" that is only for logging purposes?

Reply:
You are correct, I think using the header terminology was the incorrect one. let me fix that.

It is literally just a call option for logging purposes that the `retryMiddleware` sets for the logging logic to have this value.